x509/cert_validations.go

```diff
@@ -46,28 +48,43 @@ func validateCertChain(certChain []*x509.Certificate, expectedLeafEku x509.ExtKe
 		if signingTime != nil && (signingTime.Before(cert.NotBefore) || signingTime.After(cert.NotAfter)) {
 			return fmt.Errorf("certificate with subject %q was not valid at signing time of %s", cert.Subject, signingTime.UTC())
 		}
-
 		if err := cert.CheckSignature(cert.SignatureAlgorithm, cert.RawTBSCertificate, cert.Signature); err != nil {
```

Comment:
Do we need to validate that it was issued by itself? `isSelfSigned()`

Reply:
Here's the func doc of [CheckSignature](https://pkg.go.dev/crypto/x509#Certificate.CheckSignature). The usage here is indeed checking if it is a self-signed certificate.
Just in case you would curious about why we are not using the `isSelfSigned` here. Since the self-signed cert is the only cert in the cert chain and it's very likely not issued by CA, this cert won't pass the `CheckSignatureFrom` check, which is used in our `isSelfSigned` func. (source code of [CheckSignatureFrom)](https://cs.opensource.google/go/go/+/refs/tags/go1.20:src/crypto/x509/x509.go;l=822).